pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/segment/SegmentSizeBasedFlushThresholdUpdater.java

```diff
@@ -153,6 +169,6 @@ private long capNumRowsIfOverflow(long targetSegmentNumRows) {
     if (targetSegmentNumRows > Integer.MAX_VALUE) {
       targetSegmentNumRows = Integer.MAX_VALUE;
```

Comment:
Could this potentially cause segment generation failure because the size of index could become > 2GB? Should we ceil the value to be a much smaller one?

Reply:
This is not a part of this PR, but I agree we can choose a smaller number. Not sure how to pick a number, though. I will add a TODO for now.